### PR TITLE
feat: rewrite Optimize API auth guide

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

Part of #4117.

Rewrites the Optimize API authentication guide according to specs defined in #4117.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [x] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 1]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
+     "not-before-policy": 0
+   }
+   ```
+6. Capture the value of the `access_token` property and store it as your token.
+
+</TabItem>
+
+<TabItem value='self-managed'>
+
+1. [Configure the `api.audience` setting](/self-managed/optimize-deployment/configuration/system-configuration.md#public-api) in your Optimize installation to match the audience property of the **Optimize API** [API in Identity]($docs$/self-managed/identity/user-guide/additional-features/adding-an-api/).
```
Comment (pepopowitz) [inline optimize/apis-tools/optimize-api/optimize-api-authentication.md:60]: This is different from other API Auth guides, and I don't feel great about requesting that someone re-configure their environment in order to get an auth token....but [as discussed in Slack](https://camunda.slack.com/archives/C55U06YRG/p1726686801515669), the out-of-the-box configuration of a Self-Managed instance doesn't actually work for client_credentials authentication.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 2]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
-}
-```
-
-See the following example with _curl_:
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+All Optimize API requests except [the health readiness endpoint](./health-readiness.md) require authentication. To authenticate, generate a [JSON Web Token (JWT)](https://jwt.io/introduction/) and include it in each request.
+
+## Generating a token
```
Comment (mesellings) [inline optimize/apis-tools/optimize-api/optimize-api-authentication.md:12]:
```suggestion
## Generate a token
```
Suggestion to get away from using gerunds and make it more active, I think I suggested this in other reviews so not sure whether this was adopted - ignore if not, to keep consistent.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 3]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
+     "expires_in": 300,
+     "refresh_expires_in": 0,
+     "token_type": "Bearer",
+     "not-before-policy": 0
+   }
+   ```
+7. Capture the value of the `access_token` property and store it as your token.
+
+:::note
+The Optimize API can also be configured in a Self-Managed environment to authenticate with a single shared access token. Refer to [Public API Configuration](/self-managed/optimize-deployment/configuration/system-configuration.md#public-api) for the configuration required to access the public API using a specific token.
```
Comment (mesellings) [inline optimize/apis-tools/optimize-api/optimize-api-authentication.md:85]:
```suggestion
The Optimize API can also be configured in a Self-Managed environment to authenticate using a single shared access token. See [Public API Configuration](/self-managed/optimize-deployment/configuration/system-configuration.md#public-api) for the configuration required to access the public API using a specific token.
```

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 4]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
-  "audience": "<audience>",
-  "grant_type": "client_credentials"
-}
-```
-
-See the following example with _curl_:
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+All Optimize API requests except [the health readiness endpoint](./health-readiness.md) require authentication. To authenticate, generate a [JSON Web Token (JWT)](https://jwt.io/introduction/) and include it in each request.
```
Comment (mesellings) [inline optimize/apis-tools/optimize-api/optimize-api-authentication.md:10]:
```suggestion
All Optimize API requests except the [health readiness](./health-readiness.md) endpoint require authentication. To authenticate, generate a [JSON Web Token (JWT)](https://jwt.io/introduction/) and include it in each request.
```
Suggestion to just have the endpoint name in the link.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 5]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
+
+</TabItem>
+
+</Tabs>
+
+## Using a token
+
+Include the previously captured token as an authorization header in each request: `Authorization: Bearer <TOKEN>`.
+
+For example, to send a request to the Optimize API's ["Get dashboard IDs" endpoint](./dashboard/get-dashboard-ids.md):
```
Comment (mesellings) [inline optimize/apis-tools/optimize-api/optimize-api-authentication.md:96]:
```suggestion
For example, to send a request to the Optimize API's ["Get dashboard IDs"](./dashboard/get-dashboard-ids.md) endpoint:
```
Also, should this endpoint name be in quotes?

**Review by mesellings — APPROVED** [review]
Lgtm, just some suggestions for formatting and styling 👍

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 6]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
+<Tabs groupId="environment" defaultValue="saas" queryString values={
+[
+{label: 'SaaS', value: 'saas' },
+{label: 'Self-Managed', value: 'self-managed' },
+]}>
+<TabItem value='saas'>
+
+1. [Create client credentials]($docs$/guides/setup-client-connection-credentials/) in the **Clusters > Cluster name > API** tab of [Camunda Console](https://console.camunda.io/).
+2. Add permissions to this client for **Optimize**.
+3. Upon creating the client, capture the following values required to generate a token:
```
[the remark this answers is not in the document]
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:23]: https://github.com/camunda/camunda-docs/pull/4138#discussion_r1727666603

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:12]: Ignoring for now for consistency; I'd like to get through all these rewrites (only one more after this!), and we can standardize this later.

**Comment by pepopowitz** [comment]
@mesellings Thanks for the review! I'm going to temporarily ignore your suggestions for this PR, and leave all conversations unresolved, but add an item to #4117 to apply them to all the API auth guides. And then I'll take care of them after I finish this PR + one for the Zeebe API.

**Comment by pepopowitz** [comment]
@RomanJRW can you or someone from your team please review this PR, to confirm that I have not introduced incorrect information or steps for Optimize API authentication?

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 7]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
+   | ------------------------ | -------------------------------- | -------------------------------------------- |
+   | Client ID                | `ZEEBE_CLIENT_ID`                | -                                            |
+   | Client Secret            | `ZEEBE_CLIENT_SECRET`            | -                                            |
+   | Authorization Server URL | `ZEEBE_AUTHORIZATION_SERVER_URL` | `https://login.cloud.camunda.io/oauth/token` |
+   | Optimize REST Address    | `CAMUNDA_OPTIMIZE_BASE_URL`      | -                                            |
+   <!-- this comment convinces the markdown processor to still treat the table as a table, but without adding surrounding paragraphs. 🤷 -->
+   :::tip
+   When client credentials are created, the `Client Secret` is only shown once. Save this `Client Secret` somewhere safe.
+   :::
+4. Execute an authentication request to the token issuer:
```
[the remark this answers is not in the document]
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:35]: No preference; let me know of any specific changes you'd like me to make in a follow-up.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md [hunk 8]
```diff
@@ -4,62 +4,140 @@ title: "Authentication"
...
+   ```bash
+   curl --request POST ${ZEEBE_AUTHORIZATION_SERVER_URL} \
+       --header 'Content-Type: application/x-www-form-urlencoded' \
+       --data-urlencode 'grant_type=client_credentials' \
+       --data-urlencode 'audience=optimize.camunda.io' \
+       --data-urlencode "client_id=${ZEEBE_CLIENT_ID}" \
+       --data-urlencode "client_secret=${ZEEBE_CLIENT_SECRET}"
+   ```
+5. A successful authentication response looks like the following:
```
[the remark this answers is not in the document]
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:44]: This is a good call. Here's what it looks like currently: 

image

That number 5 could definitely be a separate paragraph under number 4. I'll adjust these across all auth guides in a follow-up.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:60]: I don't know either, but if someone does and gives me specific changes, I'll take care of them in my follow-up.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md (hunk 3 above)
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:85]: I'll adjust this across all auth guides in a follow-up.

File: optimize/apis-tools/optimize-api/optimize-api-authentication.md (hunk 5 above)
Comment (pepopowitz) [reply optimize/apis-tools/optimize-api/optimize-api-authentication.md:96]: I don't know. I can make these changes in a follow-up, if we decide they're needed.

**Review by akeller — APPROVED** [review]
🏃

**Pull request merged**